syntax = "proto2";

package RTDBM;

// Общая часть всех сообщений RTDBUS
message Header {
  required int32    protocol_version    = 1; /* версия протокола */
  required int32    exchange_id         = 2; /* системный идентификатор обмена */
  required int32    interest_id         = 3; /* идентификатор сообщения в рамках данного обмена */
  required int32    source_pid          = 4; /* системый идентификатор процесса-отправителя сообщения */
  required string   proc_dest           = 5; /* название процесса-получателя */
  required string   proc_origin         = 6; /* название процесса-создателя сообщения */
  required int32    sys_msg_type        = 7; /* системный тип сообщения */
  required int32    usr_msg_type        = 8; /* клиентский тип сообщения */
  /* может занимать как 32, так и 64 бита, в зависимости от разрядности системы */
  /* NB: используем максимально возможный размер, для общения разнородных платформ */
  required int64    time_mark           = 9; /* время выдачи запроса */
}

message ExecResult {
  required int32    exec_result         = 1; /* */
  required int32    failure_cause       = 2; /* */
}

message AskLife {
  required int32    status              = 1; /* состояние процесса */
}
